### ISSUE TITLE
Parsing exceptions occur with invocation of multiprocessing

### ISSUE THREAD

**Issue by dDArchivist** [LibraryOfCongress/bagit-python#130]
When I attempt to run bagit.py with --processes option, I receive an enormous output of parsing exceptions. This does not occur when I do NOT invoke --processes. 
[python_bagit_multiprocessing_error.txt](https://github.com/LibraryOfCongress/bagit-python/files/2903144/python_bagit_multiprocessing_error.txt)

**Comment by edsu** [comment]
Interesting, it looks like you are on Windows? If you clone the repo and run the tests do they pass?

    git clone https://github.com/libraryofcongress/bagit-python.git 
    cd bagit-python
    python setup.py test

I wonder if [this difference in behavior](https://stackoverflow.com/questions/38236211/why-multiprocessing-process-behave-differently-on-windows-and-linux-for-global-o) of multiprocessing on Windows might help explain what's going on?

**Comment by dDArchivist** [comment]
Hi. Thanks for taking a look. I followed your directions by using the bash emulator in Git for Windows. I'm attaching a sample. 

[bagit_py_test.txt](https://github.com/LibraryOfCongress/bagit-python/files/3150102/bagit_py_test.txt)

**Comment by edsu** [comment]
It looks like the test suite is noticing problems too. We should really be running thr tests on windows regularly as part of our builds.

**Comment by acdha** [comment]
Yeah, back in the day there wasn't a great Windows CI option but there are several now. In addition to Travis, I know bagit-java uses https://www.appveyor.com/ and Azure Pipelines announced a free tier for open-source: https://azure.microsoft.com/en-us/blog/announcing-azure-pipelines-with-unlimited-ci-cd-minutes-for-open-source/

**Comment by tw4l** [comment]
I'm seeing the same behavior described by @ross-spencer with `--processes` with bagit-python version 1.8.1, Python 3.10.9, on a Macbook with M2 chip.

**Comment by versmar** [comment]
I ran into this issue as well, running bagit-python version 1.9.1, Python 3.11.8 on Windows 10 and Windows Server 2019.

I think it does relate to [differences in multiprocessing](https://stackoverflow.com/questions/38236211/why-multiprocessing-process-behave-differently-on-windows-and-linux-for-global-o) suggested by edsu.

My limited understanding of the problem -
In the spawned pool processes, line 47 `MODULE_NAME = "bagit" if __name__ == "__main__" else __name__`  sets MODULE_NAME to `__mp_main__`
When this is passed to `get_distribution(MODULE_NAME)` on line 52 it causes an exception, I guess because of the underscores in `__mp_main__`.
Adding `except InvalidRequirement:` to that try/except block gets rid of those errors, but the logging generated by spawned processes still doesn't work as expected. They're getting their own logger on line 49, and it doesn't have a logging.basicConfig() statement to set things up.
I think there might be a solution in logging.QueueHandler and logging.QueueListener, but haven't looked into it further.

**Comment by WhenSkiesAbove** [comment]
Maybe this is expected behavior for those who have a better sense of this than I do, but I DON'T get this error when importing bagit. I get it in Windows when using bagit.py directly with --processes, but when I use `bagit.make_bag(bag_dir, processes=2)` in my own program, it works fine. No idea why, but mentioning it here in case it helps.
